sormas-backend/src/main/java/de/symeda/sormas/backend/sample/PathogenTestFacadeEjb.java

```diff
@@ -319,62 +318,65 @@ private void onPathogenTestChanged(PathogenTestDto existingPathogenTest, Pathoge
         final Sample sample = sampleService.getByUuid(sampleUuid);
         final Case caze = sample.getAssociatedCase();
         final Contact contact = sample.getAssociatedContact();
-        final Region region;
-        final Disease disease;
-        final List<User> messageRecipients;
 
         if (caze != null) {
-            region = caze.getRegion();
-            disease = caze.getDisease();
-            messageRecipients = userService.getAllByRegionAndUserRoles(region,
+            final Region region = caze.getRegion();
+            final Disease disease = caze.getDisease();
+            final List<User> messageRecipients = userService.getAllByRegionAndUserRoles(region,
                     UserRole.SURVEILLANCE_SUPERVISOR, UserRole.CASE_SUPERVISOR);
-        } else {
-            region = contact.getRegion();
-            disease = contact.getDisease();
-            messageRecipients = userService.getAllByRegionAndUserRoles(region,
+            sendMessageOnPathogenTestChanged(existingPathogenTest, newPathogenTest, caze, contact, disease, messageRecipients);
+        }
+
+        if (contact != null){
+            final Region region = contact.getRegion() != null ? contact.getRegion() : contact.getCaze().getRegion();
+            final Disease disease = contact.getDisease() != null ? contact.getDisease() : contact.getCaze().getDisease();
+            final List<User> messageRecipients = userService.getAllByRegionAndUserRoles(region,
                     UserRole.SURVEILLANCE_SUPERVISOR, UserRole.CONTACT_SUPERVISOR);
+            sendMessageOnPathogenTestChanged(existingPathogenTest, newPathogenTest, caze, contact, disease, messageRecipients);
```

Comment:
With this solution, surveillance supervisors will receive the mail twice. The message recipients should only be built once and contain all user roles that should receive the message.